.github/actions/deploy/deployment_manager.py

```
@@ -0,0 +1,566 @@
+
+class K8sDeploymentManager:
+    """Generic Kubernetes deployment manager with apply, wait, and debug
+    capabilities."""
+
+    def __init__(self):
+        """Initialize deployment manager."""
+        pass
+
+    def run_command(self,
+                    cmd: List[str],
+                    cwd: str = None,
```

Ruff (0.14.11)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)